lib/include/jxl/encode.h

```diff
@@ -812,16 +867,18 @@ JXL_EXPORT JxlEncoderStatus JxlEncoderSetCodestreamLevel(JxlEncoder* enc,
  * using this function with lossless set to JXL_DEC_FALSE does not guarantee
  * lossy encoding, though the default set of options is lossy.
  *
- * @param options set of encoder options to update with the new mode
+ * @param frame_options set of options and metadata for this frame. Also
+ * includes reference to the encoder object.
  * @param lossless whether to override options for lossless mode
  * @return JXL_ENC_SUCCESS if the operation was successful, JXL_ENC_ERROR
  * otherwise.
  */
-JXL_EXPORT JxlEncoderStatus
-JxlEncoderOptionsSetLossless(JxlEncoderOptions* options, JXL_BOOL lossless);
+JXL_EXPORT JxlEncoderStatus JxlEncoderOptionsSetLossless(
+    JxlEncoderOptions* frame_options, JXL_BOOL lossless);
```

Comment:
Not really related to this PR, but this one is confusing and has lead to applications getting things wrong. It only sets frame options to lossless, while the image options (in particular: using xyb or not) are still lossy by default.

By having this function like this, we give the false impression that it is possible to use lossless settings after the image header has already been written and been set to use XYB.

I think it might be better to deprecate this function and instead have a global encoder object flag for lossless that has to be set when setting the basicinfo. If you want to do multi-frame images with both vardct and 'lossless' XYB, you can do that manually.

Reply:
This function only changes frame options, that are written in the frame header. AFAIK, it is possible to write both lossless and lossy frames, or any mix thereof, after the image header was written. Which setting of the image header do you mean?

Reply:
Uses_xyb is part of the image header. If it is true, then no frames can be really lossless, since all frames are in XYB then.

Reply:
Let's handle this in a different MR if needed, maybe it should indeed become a global encoding setting, also need to check why the unit tests with JxlEncoderOptionsSetLossless are passing, given that they are not touching the xyb_encoded field...

But that's unrelated to this MR, so resolving this comment